osu.Game.Rulesets.Catch/Objects/Drawable/DrawableCatchHitObject.cs

```diff
@@ -47,26 +52,32 @@ protected DrawableCatchHitObject(CatchHitObject hitObject)
 
         protected override void CheckForJudgements(bool userTriggered, double timeOffset)
         {
+            if (CheckPosition == null) return;
+
             if (timeOffset > 0)
-                AddJudgement(new Judgement { Result = CheckPosition?.Invoke(HitObject) ?? false ? HitResult.Perfect : HitResult.Miss });
+                AddJudgement(new Judgement { Result = (bool)CheckPosition?.Invoke(HitObject) ? HitResult.Perfect : HitResult.Miss });

```

Comment:
The null coalescing operator on CheckPosition can be removed, along with the cast to bool